### PR TITLE
Renamed ies_enkf_data -> ies_enkf_state

### PR DESCRIPTION
**Pull request by joakim-hove**
Part of: https://github.com/equinor/ert/pull/2412

In this PR the class `ies_enkf_data` is renamed to `ies_enkf_state` - this is to signal that this datastructure holds on to state - with (close to) application lifetime.

### CODE REVIEW

**Comment by joakim-hove** [comment]
test this please

**Comment by joakim-hove** [comment]
test ert please

File: libres/lib/CMakeLists.txt [hunk 1]
```diff
@@ -55,7 +55,7 @@ pybind11_add_module(
   # IES_ENKF
   analysis/ies/ies_enkf.cpp
   analysis/ies/ies_enkf_config.cpp
-  analysis/ies/ies_enkf_data.cpp
+  analysis/ies/ies_enkf_state.cpp
```
Comment (jondequinor) [inline libres/lib/CMakeLists.txt:58]: Is this a "iterative ensemble smoother ensemble Kalman filter state" module?

**Review by jondequinor — CHANGES_REQUESTED** [review]

File: libres/lib/CMakeLists.txt (hunk 1 above)
Comment (joakim-hove) [reply libres/lib/CMakeLists.txt:58]: What do you want changed?

The point is that the struct `ies_enkf_data` contains state about the update which has (close to) application lifetime; this long lasting state will create complexity and to highlight that is indeed state I have suggested a rename from `data` to `state`.

Comment (jondequinor) [reply libres/lib/CMakeLists.txt:58]: If it's not an iterative ensemble smoother ensemble Kalman filter state I would like it renamed to something more sensible when we're renaming it anyway.

Comment (joakim-hove) [reply libres/lib/CMakeLists.txt:58]: Well - the whole subsystem/module is already prefixed with `ies_enkf` - does not make sense to only change one of the parts to something else. Could of course rename everything - then I would suggesting removing the `enkf` part.

Comment (jondequinor) [reply libres/lib/CMakeLists.txt:58]: Right. It's just that the "data" to "state" change does not improve anything in my feeble mind. But I know that a diff has a cost, so does a PR. So weighing cost vs benefit here makes it a bit unclear whether or not we need this. 

Anyway, I'll dismiss this review shortly and let you decide.

Comment (joakim-hove) [reply libres/lib/CMakeLists.txt:58]: > It's just that the "data" to "state" change does not improve anything in my feeble mind

Well - there we disagree. `data` has a connotation in enkf context, in this particular example the word `data` is *not* used according to that connotation.

Comment (jondequinor) [reply libres/lib/CMakeLists.txt:58]: Any documentation for this so that I can read up?

Comment (joakim-hove) [reply libres/lib/CMakeLists.txt:58]: Well not really - e.g. Geirs book; but thinking of it also `state` has connotations in that literature. I'll just close this.

**Pull request closed without merging**